iterative/utils/analytics.go

```diff
@@ -0,0 +1,144 @@
+package main
+
+import (
+	"bytes"
+	"crypto/md5"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"os/exec"
+	"regexp"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/shirou/gopsutil/host"
+	"github.com/sirupsen/logrus"
+)
+
+func SystemInfo() map[string]interface{} {
+	hostStat, _ := host.Info()
+	return map[string]interface{}{
+		"os":               hostStat.OS,
+		"platform":         hostStat.Platform,
+		"platform_version": hostStat.PlatformVersion,
+	}
+}
+
+func TaskDuration(logs string) float64 {
+	regex := regexp.MustCompile(`\w{3} \d{2} \d{2}:\d{2}:\d{2}`)
+	matches := regex.FindAllString(logs, -1)
+	taskDuration := 0.0
+
+	if len(matches) > 1 {
+		layout := "Mar 07 06:43:27"
+		t1, _ := time.Parse(layout, matches[len(matches)-1])
+		t0, _ := time.Parse(layout, matches[0])
+		taskDuration = t1.Sub(t0).Seconds()
+	}
+
+	return taskDuration
+}
+
+func IsCI() bool {
+	if _, ok := os.LookupEnv("CI"); ok {
+		return true
+	}
+
+	return false
+}
+
+func version() string {
+	return "v1.0.0"
+}
+
+func TerraformVersion() string {
+	var out bytes.Buffer
+	cmd := exec.Command("terraform", "--version")
+	cmd.Stdout = &out
+
+	err := cmd.Run()
+
+	if err == nil {
+		regex := regexp.MustCompile(`v\d+\.\d+\.\d+`)
+		version := regex.FindString(out.String())
+		return version
+	} else {
+		logrus.Error("Analytics: Failed extracting terraform version")
+	}
+
+	return ""
+}
+
+func JitsuUserId() string {
+	hostStat, _ := host.Info()
+	id := hostStat.HostID
```

Comment:
A file based mechanism alternative could be anonymise the computer ID

@shcheklein @mike0sv https://github.com/orgs/iterative/teams/cml